oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -104,11 +123,7 @@ abstract static class CredentialSource implements java.io.Serializable {
 
   protected transient HttpTransportFactory transportFactory;
 
-  @Nullable protected final ImpersonatedCredentials impersonatedCredentials;
-
-  // Internal override for impersonated credentials. This is done to keep
-  // impersonatedCredentials final.
-  @Nullable private ImpersonatedCredentials impersonatedCredentialsOverride;
+  @Nullable protected ImpersonatedCredentials impersonatedCredentials;
```

Comment:
The issue with having impersonated credentials final is that the ImpersonatedCredentials object is built by passing the current credential as the SourceCredential. If we have it final, the Impersonated Credential needs to get created in the parent constructor before the child constructor runs, which means the credential object that gets passed as the source credential is not fully instantiated. As a series of steps, the exact problem is:

1. AwsCredentials constructor is called via the builder
2. constructor super() is called, going to the External account credentials constructor
3. in the parent constructor, buildImpersonatedCredentials is called
4. the ImpersonatedCredentials object is built, but the source credential passed doesn't have any of the AwsCredentials specific builder variables set (AwsSecurityCredentialsSupplier for example) because the rest of the AwsCredentials constructor logic has not run yet, we are still in the parent constructor.

Since the source credential passed to the impersonated credential isn't actually correct here we have 1 of two options if we want any child class specific builder setters
1: Set the impersonatedCredentialOverride in the child constructor and just use that, ignoring the "final" impersonatedCredentials. This is what we do for executableCredentials, but the issue is that its kind of a weird pattern that we just have the unused and incorrect impersonatedCredentials object attached to the credential, and that it means we can't do any input validation on the child credential.
2: Make impersonatedCredentials not final, and just set it when we use it the first time when retrieving an access token. This lets us get rid of the override, and makes it so the impersonatedCredential is never built with the incorrect source credential.